### PR TITLE
Make "hide deleted message" a local user setting

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: frontend/app/src/components/home/CurrentChatMessages.svelte [hunk 1]
```diff
@@ -497,7 +498,7 @@
         return firstKey;
     }
 
-    $: groupedEvents = client.groupEvents(events, user.userId, groupInner(filteredProposals)).reverse();
+    $: groupedEvents = client.groupEvents(events, user.userId, $hideDeletedStore, groupInner(filteredProposals)).reverse();
```
Comment (julianjelfs) [inline frontend/app/src/components/home/CurrentChatMessages.svelte:501]: if the `hideDeletedStore` lives inside the client, we never need to pass its value to a method on the client.

File: frontend/app/src/components/home/pinned/PinnedMessages.svelte [hunk 2]
```diff
@@ -46,7 +48,7 @@
                         messages = {
                             kind: "success",
                             data: client
-                                .groupMessagesByDate(resp.events.sort((a, b) => a.index - b.index), user.userId)
+                                .groupMessagesByDate(resp.events.sort((a, b) => a.index - b.index), user.userId, $hideDeletedStore)
```
Comment (julianjelfs) [inline frontend/app/src/components/home/pinned/PinnedMessages.svelte:51]: as above

File: frontend/app/src/components/home/thread/Thread.svelte [hunk 3]
```diff
@@ -150,7 +151,7 @@
     $: canSend = client.canReplyInThread(chat.chatId);
     $: canReact = client.canReactToMessages(chat.chatId);
     $: messages = client
-        .groupEvents([rootEvent, ...$threadEvents], client.user.userId)
+        .groupEvents([rootEvent, ...$threadEvents], client.user.userId, $hideDeletedStore)
```
Comment (julianjelfs) [inline frontend/app/src/components/home/thread/Thread.svelte:154]: as above

**Review by julianjelfs — CHANGES_REQUESTED** [review]

File: frontend/app/src/components/home/CurrentChatMessages.svelte (hunk 1 above)
Comment (megrogan) [reply frontend/app/src/components/home/CurrentChatMessages.svelte:501]: As discussed we need the components to "react" to the hideDeleted store changing

**Review by julianjelfs — APPROVED** [review]

**Pull request merged**